clang-tools-extra/clangd/unittests/InlayHintTests.cpp

```diff
@@ -1724,6 +1724,33 @@ TEST(InlayHints, RestrictRange) {
               ElementsAre(labelIs(": int"), labelIs(": char")));
 }
 
+TEST(ParameterHints, PseudoObjectExpr) {
+  Annotations Code(R"cpp(
+    struct S {
+      __declspec(property(get=GetX, put=PutX)) int x[];
+      int GetX(int y, int z) { return 42 + y; }
+      void PutX(int y) { x = $one[[y]]; } // FIXME: Undesired `x = y: y` for this ill-formed expression.
```

Comment:
This is a bit confusing, since the fact that the expression is ill-formed is not related to the FIXME.

I would suggest moving the `x = y;` to some other function (where it's not ill-formed).

(Also mention in a comment that this builds a PseudoObjectExpr here too.)

Reply:
Sorry, I'm a bit inaccurate here: The assignment expression in `PutX` is not ill-formed but would expand to a recursive call to `PutX` since we're assigning to the member with the very setter. (Of course, this would end up crashing if we [run the program](https://gcc.godbolt.org/z/e3fW3W73a).)

I think I was suggesting that this is a bad case for supporting such an extension: the assignment expression could be of a syntactic form with an inappropriate hint location though.

Reply:
It's nice to wipe out these extra hints if it won't take too many steps. The AST shows,
```
`-PseudoObjectExpr 'void'
  |- (Syntactic) BinaryOperator
  `- (Semantic) CXXMemberCallExpr 'void'
```
I wonder if it is possible to enumerate all the bad cases we don't want (e.g. Don't walk down the expression if we encounter a `BinaryOperator` in `PseudoObjectExpr`.)

Reply:
I'm not sure what other bad cases there might be, but excluding cases where the syntactic form is a `BinaryOperator` to avoid this case sounds fine to me.